src/components/SearchPanelControls.js

```diff
@@ -9,32 +9,64 @@ import InputAdornment from '@material-ui/core/InputAdornment';
 
 /** */
 export class SearchPanelControls extends Component {
+  /** */
+  constructor(props) {
+    super(props);
+
+    this.state = { search: '' };
+    this.handleChange = this.handleChange.bind(this);
+    this.submitSearch = this.submitSearch.bind(this);
+  }
+
+  /** */
+  handleChange(event) {
+    this.setState({
+      search: event.target.value,
+    });
+  }
+
+  /** */
+  submitSearch(event) {
+    const { fetchSearch, searchService } = this.props;
+    const { search } = this.state;
+    event.preventDefault();
+    fetchSearch(searchService.options.resource.id, `${searchService.id}?q=${search}`);
+  }
+
   /** */
   render() {
     const { cwId, t } = this.props;
+    const { search } = this.state;
     const id = `search-${cwId}`;
     return (
-      <div>
+      <form onSubmit={this.submitSearch}>
         <FormControl>
           <InputLabel htmlFor={id}>{t('searchInputLabel')}</InputLabel>
           <Input
             id={id}
+            onChange={this.handleChange}
+            value={search}
             endAdornment={(
               <InputAdornment position="end">
-                <IconButton aria-label={t('searchSubmitAria')}>
+                <IconButton
+                  aria-label={t('searchSubmitAria')}
+                  type="submit"
+                >
                   <SearchIcon />
                 </IconButton>
               </InputAdornment>
             )}
           />
         </FormControl>
-      </div>
+      </form>
     );
   }
 }
 
 SearchPanelControls.propTypes = {
   cwId: PropTypes.string.isRequired,
+  fetchSearch: PropTypes.func.isRequired,
+  searchService: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
```

Comment:
It looks like we're starting to get an idea of the interface of the `searchService` we're depending on (`id` & `options`).  Just thinking if this is a good time to assert that's what we expect here (happy to defer until we know more)

Reply:
Good point. I've updated the proptypes definition for `id` and `options`